### PR TITLE
Added support for MAX_OPEN_FILES environment variable #849

### PR DESCRIPTION
**Pull request by alvagante**
# Pull Request Checklist

## Related Issue

Fixes # .

## General

- [x] Update `README.md` with any necessary configuration snippets

- [x] New parameters are documented

- [x] New parameters have tests

- [x] Tests pass - `bundle exec rake validate lint spec`

### CODE REVIEW

File: manifests/enterprise.pp [hunk 1]
```diff
@@ -28,6 +28,8 @@
 #
 # @param heap_size Value of the HEAP_SIZE environment variable.
 #
+# @param max_open_files Value of the MAX_OPEN_FILES environment variable.
```
Comment (ghoneycutt) [inline manifests/enterprise.pp]: Why is the parameter added to three different classes?

Comment (alvagante) [reply manifests/enterprise.pp]: I followed the current approach for similar cases, but actually you're right, max_open_files in this case can and should not stay in package.pp. Actually if we split the sensu.erb template used both by /etc/default/sensu and /etc/default/sensu-enterprise we will be able to set ee-only parameters only in enterprise.pp

Comment (ghoneycutt) [reply manifests/enterprise.pp]: Do we need it anywhere except the enterprise class then?

Comment (alvagante) [reply manifests/enterprise.pp]: Actually, we need these params also in init.pp, besides enterprise.pp for how the module is done and tested (all spec tests for the various subclasses, in reality, test on the main class)

Comment (ghoneycutt) [reply manifests/enterprise.pp]: Do we need it in the package class then?

Comment (alvagante) [reply manifests/enterprise.pp]: Parameters used only in puppet enterprise are not needed in packages class.
FYI just pushed with the code to support other sense ee settings like JAVA_OPTS.
Have to properly test on vagrant yet. 
These changes anyway, might be obsoleted by what we decide in #851 .

**Comment by alvagante** [comment]
@ghoneycutt this should be ready, and even if it might be refactored on the lines on what's under discussion in https://github.com/sensu/sensu-puppet/pull/852 I'd not wait for it.

**Pull request merged**